classes/app/interfaces/factory.php

```diff
@@ -22,11 +25,25 @@ public function assign(): assign_factory;
 
     public function ai(): ai_factory;
 
+    /**
+     * @template T
+     * @psalm-template T
+     * @param array $items
+     * @return collection<T>
+     */
```

Comment:
Sometimes.

If you define it at the method it doesn't work probably. Somehow this feature feel like it's still in beta :sweat_smile: 
But at the ``class``, ``interface`` works ok (not perfect).

Since Jetbrains said that they support it [link to their statement](https://blog.jetbrains.com/phpstorm/2021/07/phpstorm-2021-2-release/). We might just add it for the convenient.

Maybe it work if we change PHP level to PHP 8.1+ in the setting.